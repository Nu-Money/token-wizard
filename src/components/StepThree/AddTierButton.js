import React from 'react'

const AddTierButton = props => {
  return (
    <button type="button" className="ti-AddTierButton" onClick={props.onClick}>
      <span className="ti-AddTierButton_Text">Add Tier</span>
      <span className="ti-AddTierButton_PlusIcon" />
    </button>
  )
}

export default AddTierButton
