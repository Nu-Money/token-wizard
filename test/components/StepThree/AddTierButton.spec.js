import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import AddTierButton from '../../../src/components/StepThree/AddTierButton'

configure({ adapter: new Adapter() })

describe('AddTierButton', () => {
  it(`should render AddTierButton`, () => {
    // Given
    const wrapper = renderer.create(<AddTierButton onClick={jest.fn()} />)

    // When
    const tree = wrapper.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should call onClick prop`, () => {
    // Given
    const onClick = jest.fn()
    const wrapper = mount(<AddTierButton onClick={onClick} />)

    // When
    wrapper.find(AddTierButton).simulate('click')

    // Then
    expect(onClick).toHaveBeenCalled()
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
