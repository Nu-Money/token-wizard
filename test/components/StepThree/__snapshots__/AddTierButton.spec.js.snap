// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTierButton should render AddTierButton 1`] = `
<button
  className="ti-AddTierButton"
  onClick={[MockFunction]}
  type="button"
>
  <span
    className="ti-AddTierButton_Text"
  >
    Add Tier
  </span>
  <span
    className="ti-AddTierButton_PlusIcon"
  />
</button>
`;
